sycl/doc/extensions/experimental/sycl_ext_oneapi_bindless_images/ZE_experimental_bindless_images.asciidoc

```diff
@@ -0,0 +1,621 @@
+is the creation of a bindless image on already existing USM memory, preventing
+copies, and hardware sampling can be applied to that data.
+
+This proposal aims to extend Level Zero to better align with the PI API used by
+the SYCL runtime in DPC++.
+
+
+=== Specification
+
+To enable the separation of image memory allocation from image handle
+generation, we propose two new allocation functions `zeMemAllocImageExp` and
+`zeMemAllocPitchedExp`.
+
+`zeMemAllocImageExp` will allocate non-USM image memory with an implementation
+specific layout (such as tile swizzle patterns or lossless compression).
+
+Whereas, `zeMemAllocPitchedExp` will allocate USM memory in a linear layout.
+Memory allocated with this API should be freed using `zeMemFree`.
+
+Alongside these allocation functions, we propose a singular additional freeing
+function, `zeMemFreeImageExp`, to free memory allocated with
+`zeMemAllocImageExp`.
+
+We also propose a new struct `ze_device_image_bindless_exp_desc_t` that
+describes device restrictions on image properties. It should be passed to
+`ze_device_properties_t::pNext`. It contains the fields `imagePitchAlign`,
+`maxImageLinearWidth`, `maxImageLinearHeight`, and `maxImageLinearPitch`.
+
+== Definitions
+
+```cpp
+typedef uint64_t ze_image_mem_handle_exp_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Describes memory to be allocated on device intended for images with
+///        a device specific memory layout
+///
+/// @details
+///     - This structure is passed to:
+///       + ::zeMemAllocImageExp
+///       + ::ze_image_desc_t.pNext
+///         - When this structure is passed to an image descriptor,
+///           ::zeImageCreate must not allocate any additional device memory for
+///           the created image, only return a handle.
+///         - Additionally, when ::zeImageDestroy is called on an image created
+///           from a ::ze_image_mem_alloc_exp_desc_t allocation,
+///           ::zeImageDestroy must not free the memory associated with the
+///           image handle.
+///
+typedef struct _ze_image_mem_alloc_exp_desc_t {
+    ze_structure_type_t stype;      // [in]
+    void *pNext;                    // [in,out][optional]
+    size_t widthInBytes;            // [in]
+    size_t height;                  // [in][optional]
+    size_t depth;                   // [in][optional]
+    uint32_t elementSizeByte;       // [in]
+} ze_image_mem_alloc_exp_desc_t;
+
+///////////////////////////////////////////////////////////////////////////////
+/// @brief Describes linear USM to be allocated on device intended for images
+///
+/// @details
+///     - This structure is passed to:
+///       + ::zeMemAllocPitchedExp
+///         - ::rowPitch and ::slicePitch are [out] parameters
+///       + ::ze_image_desc_t.pNext
+///         - ::rowPitch and ::slicePitch are [in] parameters
+///         - When this structure is passed to an image descriptor,
```

Comment:
To my understanding, `zeImageCreate` will require the data fields in this struct. That said, it will also need the output from `zeMemAllocPitchedExp` which is an issue we will address in the next revision.

In the next revision, we hope to remove the image memory descriptors we've proposed and piggy-back off `ze_image_desc_t`, using `pNext` for any additional data fields we may require. This is a possible place where we could add the returned memory handle of `zeMemAllocPitchedExp`, so that it could be propogated forward to `zeImageCreate`.